Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -25,6 +25,21 @@ public class SdkHttpRequest {
         self.endpoint = endpoint
         self.body = body
     }
+    
+    public func toBuilder() -> SdkHttpRequestBuilder {
+        let builder = SdkHttpRequestBuilder()
+            .withBody(self.body)
+            .withMethod(self.method)
+            .withHeaders(self.headers)
+            .withPath(self.path)
+            .withHost(self.host)
+            .withPort(self.endpoint.port)
+            .withProtocol(self.endpoint.protocolType ?? .https)
+        if let qItems = self.queryItems {
+            builder.withQueryItems(qItems)
+        }
+        return builder
```

Comment:
Does passing an empty list cause problems? We could do instead:

```
return SdkHttpRequestBuilder()
            .withBody(self.body)
            .withMethod(self.method)
            .withHeaders(self.headers)
            .withPath(self.path)
            .withHost(self.host)
            .withPort(self.endpoint.port)
            .withProtocol(self.endpoint.protocolType ?? .https)
            .withQueryItems(qItems ?? [])
```

Reply:
That's actually what I did originally, but form Josh's comment having empty query items and having nil for query items is different - so I changed it to current state.